### ISSUE TITLE
`PackageOrganizer>>#listAtCategoryNamed:` deprecated in Pharo 12

### ISSUE THREAD

**Issue by fniephaus** [hpi-swa/smalltalkCI#618]
The smalltalkCI tests [have started to fail on Pharo 12](https://github.com/hpi-swa/smalltalkCI/actions/runs/6990959110/job/19020997366) due to a deprecation:

```
Deprecation: The method PackageOrganizer>>#categories called from SmalltalkCI class>>#allCategoryNames has been deprecated. The manipulation of categories got deprecated in Pharo 12. It is better to manipulate directly the packages and tags.
PackageOrganizer(Object)>>deprecated:
PackageOrganizer>>categories
SmalltalkCI class>>allCategoryNames
SmalltalkCI class>>classesForCategories:
SmalltalkCITest>>testClassesForCategories
SmalltalkCITest(TestCase)>>performTest
```

@theseion Could you (or someone else from the Pharo team) please take a look and fix this?

**Comment by fniephaus** [comment]
It seems https://github.com/hpi-swa/smalltalkCI/pull/609 has addressed the main issue, but the tests were not updated to use `platformClass`. Even when doing that change, I see a number of test failures on Pharo 12 (see below). Something must not be working correctly :(

```
SCIPharoMonticelloLoadSpecTest
 ✗ #testAddLoadedClassesFrom (4ms)
TestFailure: Denial failed
SCIPharoMonticelloLoadSpecTest(TestAsserter)>>assert:description:resumable:
SCIPharoMonticelloLoadSpecTest(TestAsserter)>>deny:description:resumable:
SCIPharoMonticelloLoadSpecTest(TestAsserter)>>deny:description:
SCIPharoMonticelloLoadSpecTest(TestAsserter)>>deny:
SCIPharoMonticelloLoadSpecTest>>testAddLoadedClassesFrom ...deny: (self spec loadedClasses isEmpty)
SCIPharoMonticelloLoadSpecTest(TestCase)>>performTest

SmalltalkCISpecTest
 ✗ #testClassesToTest (7ms)
TestFailure: Assertion failed
SmalltalkCISpecTest(TestAsserter)>>assert:description:resumable:
SmalltalkCISpecTest(TestAsserter)>>assert:description:
SmalltalkCISpecTest(Object)>>assert:
SmalltalkCISpecTest>>testClassesToTest ...assert: (classes includes: SmalltalkCITest)
SmalltalkCISpecTest(TestCase)>>performTest

SmalltalkCITest
 ✗ #testClassesForCategories (3ms)
TestFailure: Assertion failed
SmalltalkCITest(TestAsserter)>>assert:description:resumable:
SmalltalkCITest(TestAsserter)>>assert:description:
SmalltalkCITest(Object)>>assert:
SmalltalkCITest>>testClassesForCategories ...assert: (classes includes: SmalltalkCI)
SmalltalkCITest(TestCase)>>performTest
 ✗ #testClassesFrom (2ms)
TestFailure: Assertion failed
SmalltalkCITest(TestAsserter)>>assert:description:resumable:
SmalltalkCITest(TestAsserter)>>assert:description:
SmalltalkCITest(Object)>>assert:
SmalltalkCITest>>testClassesFrom ...assert: (classes includes: SmalltalkCI)
SmalltalkCITest(TestCase)>>performTest
 ✗ #testClassesInCategory (0ms)
TestFailure: Assertion failed
SmalltalkCITest(TestAsserter)>>assert:description:resumable:
SmalltalkCITest(TestAsserter)>>assert:description:
SmalltalkCITest(Object)>>assert:
SmalltalkCITest>>testClassesInCategory ...assert: (classes includes: SmalltalkCI)
SmalltalkCITest(TestCase)>>performTest
 ✗ #testClassesWithCategoryNames (1ms)
TestFailure: Assertion failed
SmalltalkCITest(TestAsserter)>>assert:description:resumable:
SmalltalkCITest(TestAsserter)>>assert:description:
SmalltalkCITest(Object)>>assert:
SmalltalkCITest>>testClassesWithCategoryNames ...assert: (classes includes: SmalltalkCI)
SmalltalkCITest(TestCase)>>performTest

SCITestReporterXMLTest
 ✗ #testReportSuccess (20ms)
TestFailure: Assertion failed
SCITestReporterXMLTest(TestAsserter)>>assert:description:resumable:
SCITestReporterXMLTest(TestAsserter)>>assert:description:
SCITestReporterXMLTest(Object)>>assert:
SCITestReporterXMLTest>>testReportSuccess ...assert: index > 0
SCITestReporterXMLTest(TestCase)>>performTest

SCITestRunnerTest
 ✗ #testRunClasses (14ms)
TestFailure: Assertion failed
SCITestRunnerTest(TestAsserter)>>assert:description:resumable:
SCITestRunnerTest(TestAsserter)>>assert:description:
SCITestRunnerTest(Object)>>assert:
SCITestRunnerTest>>testRunClasses ...assert: runner isSuccessful
SCITestRunnerTest(TestCase)>>performTest
 ✗ #testRunSpecNamed (11ms)
TestFailure: Assertion failed
SCITestRunnerTest(TestAsserter)>>assert:description:resumable:
SCITestRunnerTest(TestAsserter)>>assert:description:
SCITestRunnerTest(Object)>>assert:
SCITestRunnerTest>>testRunSpecNamed ...assert: runner isSuccessful
SCITestRunnerTest(TestCase)>>performTest
```

**Comment by theseion** [comment]
Checking...

**Comment by theseion** [comment]
This has been fixed in #619.
